drivers/rtc/rtc_ll_stm32.c

```diff
@@ -81,7 +81,7 @@ static int rtc_stm32_enter_initialization_mode(bool kernel_available)
 {
 	if (kernel_available) {
 		LL_RTC_EnableInitMode(RTC);
-		bool success = WAIT_FOR(LL_RTC_IsActiveFlag_INIT(RTC), RTC_TIMEOUT, k_msleep(1));
+		bool success = WAIT_FOR(LL_RTC_IsActiveFlag_INIT(RTC), RTC_TIMEOUT, k_busy_wait(1));
```

Comment:
The point of using WAIT_FOR and k_msleep was to avoid a busy wait, so it is probably better to reduce this to just call LL_RTC_EnterInitMode() even if kernel is available.